### DISCUSSION TITLE
Upgrade / Downgrade and Printing issues.

### DISCUSSION THREAD

**Discussion by tuxpeople** [Jyers/Marlin#1306] [Q&A] [answered]
Hi guys

After spending many hours to fix my issues myself, I need your help. It may be worth mentioning that I have the original Creality BLTouch Kit. installed on my Ender 3v2. It's all original except for the nozzle (is a 0.4 micro swiss now), new PTFE tube and better bed springs.

Let me go back and give you some history:

From a hint in a youtuber's video (I think it was CHEP), I started using Smiths3D Firmware for my Ender 3v2. Very successful. Last time a came across [their site](https://smith3d.com/ender-3-v2-bltouch-firmware-installation-guide-by-smith3d-com/) to look for an upgrade, I discovered that they discontinued their Marlin build in favor of this one here. So I upgraded to Jyers `E3V2-BLTouch-3x3-v4.2.2-v1.3.4.bin`. But I then ran into the "Kill Reason M112 - Probe error" issue as described in #192. I solved that by upgrading from `E3V2-BLTouch-3x3-v4.2.2-v1.3.4.bin` to `E3V2-BLTouch-3x3-v4.2.2-v1.3.5.bin`. I configured X and Y Probe Offsets  

But I didn't managed to get a proper print done. I used the very same gcode as before. But the print was unclean, layers not sticking together:
![IMG_8229](https://user-images.githubusercontent.com/1506076/129357649-30705afb-3a20-4719-900c-c85a1164d3af.jpg)

I tried to fix that. I ended up having initial layers like this:
![IMG_8236](https://user-images.githubusercontent.com/1506076/129357946-06bd47ed-bccd-4ed6-9325-a2a3e3c61eea.jpg)

A friend of mine, more experienced with 3d printing than I am, said it could be that my Z-Offset is wrong. So I put a paper on the bed, did a Z-Homing. I wasn't able to move the paper at all. So I tried to calibrating my Z-Offset. Enabled live adjustments, and set the the z-offset to being able to move the paper under the nozzle, while it was scratching a bit on the paper. I then started a print. All I got was spaghetti. Once again,  put a paper on the bed, did a Z-Homing. Paper was moving freely. With the very same settings as it scratched a bit before. I endet up setting the Z-Offset back almost to the same value I had before. Started another print. No spaghetti, but layers as on above pictures.

I wasn't able to fix that. Coming back here, I saw the new release `E3V2-BLTouch-3x3-v4.2.2-v1.3.5.b.bin`, updated to it. No luck either. Layers looked like this:
![IMG_8330](https://user-images.githubusercontent.com/1506076/129359128-7cebcbd2-6903-415c-84d0-553ffdbb7b94.jpg)

So I gave up. I tried to download the last release of the Smith3D-Marlin build. Put it on the SD card. Booted the printer. The screen went on, but stayed black. I tried many different firmware releases from here, from Smith3d, from Creality. No luck.

During my research, I ended up here:
https://forum.creality.com/topic/94/ender-3-v2-black-screen/9?_=1628846282347

When I download that `DWIN_SET(3).zip` and the `Ender-3 V2-Marlin-2.0.1 - V1.0.1 .bin` I was able to bring my printer back to live using this bin and the content of the ZIP on my SD card. But I'm not able to change to any other firmware now. 

Does someone has a hint for me? Actually, what I would like is to go back to Jyers firmware and get decent prints again :-)

**Update:**
No idea how, but after trying it over and over again, I'm now back on `E3V2-BLTouch-3x3-v4.2.2-v1.3.5.bin`. Does anyone have an idea why print quality is so bad with Jyers? Most probably I'm doing something wrong.

**Comment by kted** [comment, marked as answer]
First off, regarding flashing problems, remember that in order to flash a firmware, the .bin file must have a different filename from the one you flashed last. It also helps to have a freshly FAT32 formatted card with only the .bin file in it.

From your pictures, I'd say you have the Z-offset set incorrectly.
Also, you should have reset all settings to factory defaults after flashing, to clear any settings from the previous firmware.
